.github/workflows/ci.yml

```diff
@@ -15,10 +15,27 @@ on:
 
 jobs:
 
-  CI:
-    uses: ./.github/workflows/merge-gate.yml
-    secrets: inherit
+  checks:
+    name: Checks
+    uses: ./.github/workflows/checks.yml
+
+  Tests:
+    name: Tests
+    uses: ./.github/workflows/tests_with_coverage.yml
+
+  # This job ensures inputs have been executed successfully.
+  approve-merge:
+    name: Allow Merge
+    runs-on: ubuntu-latest
+    # If you need additional jobs to be part of the merge gate, add them below
+    needs: [ Tests ]
```

Comment:
```suggestion
    needs: [Checks, Tests ]
```

Reply:
Alternative you can also make Approval depend on metrics and metrics depends on tests and checks

Reply:
Did the second option.